### ISSUE TITLE
Narrow theme requirements

### ISSUE THREAD

**Issue by CarloPalinckx** [MyOnlineStore/bricks#325]
### Is your feature request related to a problem? Please describe ⚡️
Right now the theme requires way too much setup for users, even if we would have a partial solution for this with #41. Every slight theme change would still result into a breaking change because all properties are required. So even new additions would break builds.

### Describe the solution you'd like ✨
Maybe we could consider narrowing required properties for the theme object that we can derive other properties from. This could be a set of maybe something like 10 required fields. Maybe think along the lines of: 

```ts
type ThemeType = {
    contrastColor: string;
    colors: {
        primary: string;
        secondary: string;
        darkText: string;
        lightText:string;
        darkHeading: string;
        lightHeading:string;
    };
    fonts: {
        text: string;
        heading:string;
    };
    fontsizes?: { // We could provide a default set here.
        text?: number;
        heading?: // We could calculate missing heading sizes here
            | [number]
            | [number, number]
            | [number, number, number]
            | [number, number, number, number]
            | [number, number, number, number, number]
            | [number, number, number, number, number, number]
    };
    roundness: number;
    overrides?: RecursivePartial<ComponentsThemeType>;
}
```

We could use chroma.js to turn these into the full fledged theme.

Would love some feedback on this!

**Comment by CarloPalinckx** [comment]
I am going to be editing the list in the issue with all the properties we gather ⤴️

**Comment by Dreamdealer** [comment]
As far as I know, all font sizes (text and headings) are calculable.

**Comment by hborghols** [comment]
I totally agree that the theme is way too big to set-up so I think this is a good idea. I'll try to look into which properties I think should be required later today.

**Comment by CarloPalinckx** [comment]
> Wait, this is for the storefronts? Then discard my comment.

Bricks does not correspond to any specific product. Every possible product should be build-able with this.

**Comment by hborghols** [comment]
I actually think your list is pretty complete. The only thing I'm missing that's currently themeable is `roundness`. There are still quite a lot of rgba colors with opacity in the theme file. I can convert those to theme colors (if applicable) using the hex converter somewhere this week.

**Comment by woutsluiter** [comment]
This would definitely make theming more easy! Though I think we should carefully forge the list of basic props. A new addition here would still be breaking. Unless you'd deep-merge these values with a default/fallback set. Then it would at least be able compile and run.

**Comment by hborghols** [comment]
I agree but I think this list is already heading in a direction where it would not break very often as most new theme attributes for new components could be derived from this list. Are there specific scenario’s you’re worried about?

**Comment by woutsluiter** [comment]
The problem of #311 would still apply with the overrides if something changes in `ComponentsThemeType`, so a fallback / deepmerge would still be necessary if it's not going to be breaking.

Scenario I'm worried about is that we encounter problems with theming the new storefronts and having to make breaking changes again. But that might be looking to far ahead. Maybe store/broad theming could be one of the goals for a `2.x` version. I'll also have a look at required theming props.

**Comment by CarloPalinckx** [comment]
> The problem of #311 would still apply with the overrides 

We should definitely deepmerge the overrides on the theme that is produced by this slimmed down theme. Otherwise it wouldn't really be overrides I guess 😅

**Comment by hborghols** [comment]
Ah of course, didn't think of the overrides, good point 😄

**Comment by CarloPalinckx** [comment]
There is work being done on this in #358

**Comment by CarloPalinckx** [comment]
This was resolved with the merge of #358
